libcxx/include/__type_traits/is_nothrow_destructible.h

```diff
@@ -24,7 +21,12 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
-#if !defined(_LIBCPP_CXX03_LANG)
```

Comment:
Is there a reason why we did it differently in C++03 and non-C++03?

Reply:
Yes. The non-C++03 version uses `noexcept()`, which isn't available in C++03.

Reply:
So.. do I understand correctly that the new version simply doesn't work at all in C++03 since it tries using `noexcept` on line 39?

Reply:
The change is that trivially destructible types and types where the destructor is marked `throw()` are now considered nothrow destructible now.

Reply:
Ah, I get it now. Under Clang C++03, we used to take the `#else`. Now we'll take the code path that uses the builtin and we'll find that more things are nothrow destructible. This LGTM, but the tests should change (or be improved) to reflect this.